wd_sched.c

```diff
@@ -197,10 +197,12 @@ static handle_t session_sched_init(handle_t h_sched_ctx, void *sched_param)
 			return (handle_t)(-WD_ENOMEM);
 		}
 	} else {
-		skey->type = param->type;
 		skey->numa_id = param->numa_id;
 	}
 
+	if (param)
+		skey->type = param->type;
```

Comment:
Your modification here is not complete. I have a patch that needs to be added, and I will deal with your modification together.